### DISCUSSION TITLE
Update multiple index on one Nominatim instance

### DISCUSSION THREAD

**Discussion by felix-geovelo** [komoot/photon#796] [Q&A]
Hello !

I want to keep my index up-to-date but I serve my geocoder with redundancy.
- I've built a photon index with my Nominatim instance.
- I use this index on multiple docker-hosted geocoder

Is it possible, at the start of the container (after downloading the initial photon index), to do the following start-up (with a cron job triggering update) : 
`java -jar photon-*.jar -database nominatim -user nominatim -password ... -nominatim-update-init-for update_user`
`java -jar photon-*.jar -enable-update-api -database nominatim -user nominatim -password ... `
Having multiple container doing this at the same time will broke all indexes ? or make some of them miss updates ? Or is it totally safe ?

Thanks !

**Comment by lonvia** [comment]
You cannot run updates from the same Nominatim instance on multiple Photon instances. Photon keeps state in the Nominatim database that tracks the progress of the updates and makes sure they are applied exactly once.

If you want a distributed Photon instance, set up an external ElasticSearch with multiple distributed nodes and make Photon use that through the `-transport-address` parameters.

**Reply by felix-geovelo** [reply]
Ok, thanks for the quick response !

**Reply by felix-geovelo** [reply]
I struggle a bit implementing this solution.
I've set up two ES instance on two different server with distinct IP, both are reachable and launching a -nominatim-import on one at a time is working but i can't launch a -nominatim-import on both at the same time (giving the two address, comma separated, in the -transport-addresses param during -nominatim-import)
The error photon gives me in this case :
[photon/1qFmKm0oShCqPLf2LZ6xLQ] ResourceAlreadyExistsException[index [photon/1qFmKm0oShCqPLf2LZ6xLQ] already exists

Am I supposed to filled them one at a time ? And only use both ip when lauching photon server ?

Thanks
